x-pack/plugins/monitoring/deprecations.js

```diff
@@ -29,12 +30,19 @@ export const deprecations = ({ rename }) => {
       delete settings.elasticsearch.ssl.verify;
 
       log('Config key "xpack.monitoring.elasticsearch.ssl.verify" is deprecated. ' +
-          'It has been replaced with "xpack.monitoring.elasticsearch.ssl.verificationMode"');
+        'It has been replaced with "xpack.monitoring.elasticsearch.ssl.verificationMode"');
     },
     (settings, log) => {
       if (has(settings, 'report_stats')) {
         log('Config key "xpack.monitoring.report_stats" is deprecated and will be removed in 7.0. ' +
-            'Use "xpack.xpack_main.telemetry.enabled" instead.');
+          'Use "xpack.xpack_main.telemetry.enabled" instead.');
+      }
+    },
+    (settings, log) => {
+      const clusterAlertsEnabled = get(settings, 'cluster_alerts.enabled');
+      const emailNotificationsEnabled = clusterAlertsEnabled && get(settings, 'cluster_alerts.email_notifications.enabled');
+      if (emailNotificationsEnabled && !get(settings, CLUSTER_ALERTS_ADDRESS_CONFIG_KEY)) {
+        log(`Config key "${CLUSTER_ALERTS_ADDRESS_CONFIG_KEY}" will be required for email notifications to work in 7.0."`);
```

Comment:
I'm running this PR branch and trying to get this log to show up, but I'm not seeing it. I see `undefined` for `clusterAlertsEnabled`, `emailNotificationsEnabled`. I have a trial license and cluster alerts are enabled

Reply:
I discovered while putting this PR together that the depreciation system only passes through the settings that are explicitly defined in your `kibana.yml`. Settings that are defaulted via the plugin schema are not provided.

To say another way, you'll see this warning if and only if you define the following in your `kibana.yml`:
```yml
xpack.monitoring.cluster_alerts.enabled: true
xpack.monitoring.cluster_alerts.email_notifications.enabled: true
```

![image](https://user-images.githubusercontent.com/3493255/44752816-37cc3880-aaea-11e8-9d05-aef98664efa5.png)


This is what led me to implement the [other deprecation warning](https://github.com/elastic/kibana/pull/22195/files/ca5a8a1e47c657ecae3c4383f254003457a8b2ce#diff-df4ab00a203d84a1b958773f587c2c28R35), so that it would be more obvious/helpful to ordinary installations that don't redefine default configuration settings.

Reply:
I discovered while putting this PR together that the depreciation system only passes through the settings that are explicitly defined in your `kibana.yml`. Settings that are defaulted via the plugin schema are not provided.

To say another way, you'll see this warning if and only if you define the following in your `kibana.yml`:
```yml
xpack.monitoring.cluster_alerts.enabled: true
xpack.monitoring.cluster_alerts.email_notifications.enabled: true
```

This is what led me to implement the [other deprecation warning](https://github.com/elastic/kibana/pull/22195/files/ca5a8a1e47c657ecae3c4383f254003457a8b2ce#diff-df4ab00a203d84a1b958773f587c2c28R35), so that it would be more obvious/helpful to ordinary installations that don't redefine default configuration settings.

Reply:
Good to know, thanks!